panda/src/audiotraits/openalAudioManager.cxx

```diff
@@ -117,10 +118,11 @@ OpenALAudioManager() {
 
   // Initialization
   if (_active_managers == 0 || !_openal_active) {
-    _device = alcOpenDevice(NULL); // select the "preferred device"
+    string dev_name = get_audio_device();
+    _device = alcOpenDevice(dev_name.c_str()); // select the user or preferred device
```

Comment:
Instead of `NULL` you're now passing the empty string if the "default device" is selected.